server/routes.go

```diff
@@ -159,15 +159,20 @@ func GenerateHandler(c *gin.Context) {
 	}
 
 	// validate the request
+	isContextSet := req.Context != nil && len(req.Context) > 0
+	areMessagesSet := req.Messages != nil && len(req.Messages) > 0
```

Comment:
Instead of creating this boolean I think it's fine to just have `len(req.Messages) > 0` in the checks below. `len(nil) == 0`